nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/FormatUtils.java

```diff
@@ -425,4 +436,40 @@ public static String formatNanos(final long nanos, final boolean includeTotalNan
 
         return sb.toString();
     }
+
+    public static DateTimeFormatter prepareLenientCaseInsensitiveDateTimeFormatter(String pattern) {
+        return new DateTimeFormatterBuilder()
+                .parseLenient()
+                .parseCaseInsensitive()
+                .appendPattern(pattern)
+                .toFormatter(Locale.US);
+    }
+
+    /**
+     * Parse text to Instant - support different formats like: zoned date time, date time, date, time (similar to those supported in SimpleDateFormat)
+     * @param formatter configured formatter
+     * @param text      text which will be parsed
+     * @return parsed Instant
+     */
+    public static Instant parseInstant(DateTimeFormatter formatter, String text) {
```

Comment:
`parseToInstant()` might be a better name because the method does not parse an Instant but it parses text to Instant. Though I do not insist on it.